src/lib/Scenes/Artwork/Components/CommercialButtons/BidButton.tsx

```diff
@@ -0,0 +1,153 @@
+import { Button, Sans } from "@artsy/palette"
+import { BidButton_artwork } from "__generated__/BidButton_artwork.graphql"
+import SwitchBoard from "lib/NativeModules/SwitchBoard"
+import React from "react"
+import { createFragmentContainer, graphql, RelayProp } from "react-relay"
+
+export const PREDICTION_URL = "https://live-staging.artsy.net"
+
+export interface BidButtonProps {
+  artwork: BidButton_artwork
+  relay: RelayProp
+}
+
+export class BidButton extends React.Component<BidButtonProps> {
+  setMaxBid = (newVal: number) => {
+    this.setState({ selectedMaxBidCents: newVal })
+  }
+
+  handleBid = () => {
+    SwitchBoard.presentModalViewController(this, `/bid/${this.props.artwork.slug}`)
+  }
+
+  redirectToRegister = () => {
+    const { sale } = this.props.artwork
+    SwitchBoard.presentModalViewController(this, `/auction-registration/${sale.slug}`)
```

Comment:
These `presentModalViewController` calls should be `presentNavigationViewController` (Eigen knows to present the right controllers modally, and to present the others as pushes.)

Reply:
Cool updating.